planner/core/stats.go

```diff
@@ -463,13 +543,20 @@ func (ds *DataSource) buildIndexMergeOrPath(partialPaths []*util.AccessPath, cur
 }
 
 // DeriveStats implement LogicalPlan DeriveStats interface.
-func (p *LogicalSelection) DeriveStats(childStats []*property.StatsInfo, selfSchema *expression.Schema, childSchema []*expression.Schema) (*property.StatsInfo, error) {
+func (p *LogicalSelection) DeriveStats(childStats []*property.StatsInfo, selfSchema *expression.Schema, childSchema []*expression.Schema, _ [][]*expression.Column) (*property.StatsInfo, error) {
+	if p.stats != nil {
+		return p.stats, nil
+	}
 	p.stats = childStats[0].Scale(SelectionFactor)
+	p.stats.GroupNDVs = nil
```

Comment:
How about using the selectivity to adjust the NDV like what you do in `StatsInfo.Scale(factor)`?

Reply:
Why set it to Nil here? It seems reasonable to adjust(scale) the NDV by `SelectionFactor`.

Reply:
Unlike the DataSource case, the magic number `0.8` here utilizes no information from the filter at all, scaling the GroupNDV with this SelectionFactor may incur estimation errors IMHO, so I reset the GroupNDVs to empty for selection operator.